packages/grid/x-data-grid/src/utils/fastShallowCompare.ts

```diff
@@ -0,0 +1,34 @@
+const is = Object.is;
+
+export function fastShallowCompare<T extends Record<string, unknown> | null | undefined>(
+  a: T,
+  b: T,
```

Comment:
I feel like we shouldn't assume that both values are of the same type. In some cases, we might compare `object` and `null`/`undefined`

Reply:
The type constraint on T is `extends Record<string, unknown> | null | undefined`, IIUC `T` is at most `Record<string, unknown> | null | undefined`.

![image](https://github.com/mui/mui-x/assets/1423607/c3fd0ce4-5a41-4221-878f-1672ed342680)

Reply:
Yes, you're right 👍